odoo/addons/base/models/ir_sequence.py

```diff
@@ -53,8 +53,14 @@ def _select_nextval(cr, seq_name):
 
 def _update_nogap(self, number_increment):
     number_next = self.number_next
-    self._cr.execute("SELECT number_next FROM %s WHERE id=%%s FOR UPDATE NOWAIT" % self._table, [self.id])
-    self._cr.execute("UPDATE %s SET number_next=number_next+%%s WHERE id=%%s " % self._table, (number_increment, self.id))
+    try:
+        self._cr.execute("SELECT number_next FROM %s WHERE id=%%s FOR UPDATE NOWAIT" % self._table, [self.id])
+        self._cr.execute("UPDATE %s SET number_next=number_next+%%s WHERE id=%%s " % self._table, (number_increment, self.id))
+    except OperationalError as e:
+        if e.pgcode == '55P03':
+            raise UserError(_("This transaction wasn't executed because another transaction is already using the same no-gap sequence. Please try again later."))
+        else:
+            raise e
```

Comment:
When re-raising the current exception, use a bare `raise`: not sure that's still the case but an explicit `raise` could have odd side-effects (e.g. reset the traceback) which is undesirable.